eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/Primitives/KnownManagementTypes.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources.Models;
+using Microsoft.TypeSpec.Generator.Primitives;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Azure.Generator.Mgmt.Primitives
+{
+    internal class KnownManagementTypes
+    {
+        private static readonly IReadOnlyDictionary<string, CSharpType> _idToTypeMap = new Dictionary<string, CSharpType>()
+        {
+            ["Azure.ResourceManager.CommonTypes.ExtendedLocation"] = typeof(ExtendedLocation),
+            ["Azure.ResourceManager.CommonTypes.ExtendedLocationType"] = typeof(ExtendedLocationType),
+            ["Azure.ResourceManager.CommonTypes.ManagedServiceIdentity"] = typeof(ManagedServiceIdentity),
+            ["Azure.ResourceManager.CommonTypes.ManagedServiceIdentityType"] = typeof(ManagedServiceIdentityType),
+            ["Azure.ResourceManager.CommonTypes.OperationStatusResult"] = typeof(OperationStatusResult),
+            ["Azure.ResourceManager.CommonTypes.ProxyResource"] = typeof(ResourceData),
+            ["Azure.ResourceManager.CommonTypes.Resource"] = typeof(ResourceData),
+            ["Azure.ResourceManager.CommonTypes.SystemData"] = typeof(SystemData),
+            ["Azure.ResourceManager.CommonTypes.TrackedResource"] = typeof(TrackedResourceData),
+            ["Azure.ResourceManager.CommonTypes.UserAssignedIdentity"] = typeof(UserAssignedIdentity),
+        };
+
+        private static readonly HashSet<CSharpType> _knownTypes = _idToTypeMap.Values.ToHashSet(new CSharpFullNameComparer());
+
+        public static bool IsKnownManagementType(CSharpType type) => _knownTypes.Contains(type);
+
+        public static bool TryGetManagementType(string id, [MaybeNullWhen(false)] out CSharpType type) => _idToTypeMap.TryGetValue(id, out type);
+
+        // We only care about the namesapce and name for known types
+        private class CSharpFullNameComparer : IEqualityComparer<CSharpType>
```

Comment:
shall we add some comments to describe why we have to need this and the `Equals` on CSharpType cannot work?

Reply:
added comments